src/openjd/adaptor_runtime/_background/frontend_runner.py

```diff
@@ -107,27 +126,41 @@ def init(
             ]
         else:
             args = [str(reentry_exe)]
+
         args.extend(
             [
                 "daemon",
                 "_serve",
-                "--connection-file",
-                self._connection_file_path,
                 "--init-data",
                 json.dumps(init_data),
                 "--path-mapping-rules",
                 json.dumps(path_mapping_data),
+                "--connection-file",
+                str(connection_file_path),
             ]
         )
+
+        bootstrap_id = uuid.uuid4()
+        bootstrap_log_dir = tempfile.gettempdir()
+        bootstrap_log_path = os.path.join(
+            bootstrap_log_dir, f"adaptor-runtime-background-bootstrap-{bootstrap_id}.log"
+        )
+        args.extend(["--log-file", bootstrap_log_path])
+
+        _logger.debug(f"Running process with args: {args}")
+        bootstrap_output_path = os.path.join(
+            bootstrap_log_dir, f"adaptor-runtime-background-bootstrap-output-{bootstrap_id}.log"
+        )
+        output_log_file = open(bootstrap_output_path, mode="w+")
         try:
             process = subprocess.Popen(
                 args,
                 shell=False,
                 close_fds=True,
                 start_new_session=True,
                 stdin=subprocess.DEVNULL,
-                stdout=subprocess.DEVNULL,
-                stderr=subprocess.DEVNULL,
+                stdout=output_log_file,
```

Comment:
Being paranoid... does this prevent the shell that runs the `daemon init` command from exiting at all? I'm guessing not, but it would be good to confirm. (e.g. run `adaptor daemon init` in a shell script and make sure that the script exits, leaving the daemon running)

Reason that I ask is that this will result in a file descriptor from the foreground process being shared into the daemon.

Reply:
Confirmed the shell script exits after running daemon start: (updated my test adaptor to actually print something)

```
$ cat start-daemon.sh 
#!/bin/bash

fake-adaptor daemon start
$ ./start-daemon.sh && echo "hello"
INFO: Applying user-level configuration: /home/jericht/.openjd/adaptors/runtime/configuration.json
INFO: Applying user-level configuration: /home/jericht/.openjd/adaptors/FakeAdaptor/FakeAdaptor.json
INFO: Initializing backend process...
INFO: Started backend process. PID: 23578
INFO: Waiting for File '/tmp/ojd-ar-akzydsg1/connection.json' to exist
INFO: Retrying in 1s...
INFO: Waiting for File '/tmp/ojd-ar-akzydsg1/connection.json' to be openable
INFO: Waiting for File '/tmp/ojd-ar-akzydsg1/connection.json' to have valid ConnectionSettings
INFO: ========== BEGIN BOOTSTRAP OUTPUT CONTENTS ==========
INFO: INFO: Applying user-level configuration: /home/jericht/.openjd/adaptors/runtime/configuration.json
INFO: INFO: Applying user-level configuration: /home/jericht/.openjd/adaptors/FakeAdaptor/FakeAdaptor.json
INFO: ========== END BOOTSTRAP OUTPUT CONTENTS ==========
INFO: Checking for bootstrap logs at '/tmp/adaptor-runtime-background-bootstrap-5851ddcf-3346-4ebf-9168-b9a3a74ce235.log'
INFO: ========== BEGIN BOOTSTRAP LOG CONTENTS ==========
INFO: [2024-05-31 22:15:00,851][INFO    ] Applying user-level configuration: /home/jericht/.openjd/adaptors/runtime/configuration.json
INFO: [2024-05-31 22:15:00,853][INFO    ] Applying user-level configuration: /home/jericht/.openjd/adaptors/FakeAdaptor/FakeAdaptor.json
INFO: [2024-05-31 22:15:00,853][INFO    ] Running in background daemon mode.
INFO: ========== END BOOTSTRAP LOG CONTENTS ==========
INFO: Verifying connection to backend...
INFO: Connected successfully
openjd_env: OPENJD_ADAPTOR_SOCKET=/tmp/.openjd_adaptor_runtime/23578
ADAPTOR_OUTPUT: INFO: Running in background daemon mode.
ADAPTOR_OUTPUT: WARNING: start
nice
hello
$
```